costar_hyper/costar_hypertree_inference.py

```diff
@@ -0,0 +1,185 @@
+    def evaluate_model(self, generator, result_filename):
+        '''
+        Evaluates the initialized model and stores all metrics as per cases in cornell_grasp_train.py.
+
+        #Arguments
+        generator: Generator object for feeding data for evaluations
+        result_filename: The filename for evaluated metrics to be stored.
+        '''
+        filenames_updated, file_len_list = self.file_list_updated, self.file_len_list
+        hyperparams = load_hyperparams_json(self.hyperparams_json)
+        hyperparams.pop('checkpoint', None)
+        model = get_compiled_model(**hyperparams, problem_name=self.problem_name, load_weights=self.load_weights)
+        bsg = self.block_stacking_generator(generator)
+        with open(result_filename, 'w') as fp:
+            cw = csv.writer(fp, delimiter=',', lineterminator='\n')
+            cw.writerow(['example', 'frame_no'] + model.metrics_names)
+            # fp.write("\n")
+        frame_counter = 0
+        file_counter = 0
+        frame_len = file_len_list[file_counter]
+        for i in range(len(generator)):
+            data = next(bsg)
+            if filenames_updated[i] != filenames[file_counter]:
+                file_counter += 1
+                frame_len = file_len_list[file_counter]
+                frame_counter = 0
+
+            # print("len of X---", len(data[0]))
+            frame_counter += 1 % frame_len
+            score = model.evaluate(data[0], data[1])
+            with open("inference_results_per_frame.csv", 'a') as fp:
+                cw = csv.writer(fp, delimiter=',', lineterminator='\n')
+                score = [file_counter] + [frame_counter] + score
+                cw.writerow(score)
+
+
+    def generate_plots(self, score_file, metric_2):
+        '''
+        Generates plots for the given metric
+        #Arguments
+        score_file: name of the file containing the metrics
+        metric_2: name of metric to be used in plot generation
+        '''
+        with open(score_file, 'r') as fp:
+            reader = csv.reader(fp)
+            headers = next(reader, None)
+            scores = list(reader)
+        # metric_1_index = headers.index(metric_1)
+        metric_2_index = headers.index(metric_2)
+        frames = []
+        loss = []
+        for row in scores:
+            frames.append(row[1])
+            loss.append(row[metric_2_index])
+
+        print(headers)
+        frames = list(map(int, frames))
+        loss = list(map(float, loss))
+        figure1 = plt.figure(1, figsize=(20, 10))
+        plt.xticks(np.arange(min(frames), max(frames)+1, 10))
+        plt.yticks(np.arange(min(loss), max(loss)+1, 0.1))
+        indexes = np.where(np.array(frames) == 1)[0]
+        # print(indexes)
+        ax = plt.axes()
+        n_lines = len(indexes)
+        ax.set_color_cycle([plt.cm.cool(i) for i in np.linspace(0, 1, n_lines)])
+        count = 0
+        for i in indexes[1:]:
+            goals = self.gripper_action_goal_idx[count]
```

Comment:
I'll add more comments and clean up the code.
I wanted a single function for plotting because making a function every time we add a new plot did not make sense. Anyways, I'll split it into two for now.